### PR TITLE
Feat/validate query enhancement

### PR DESCRIPTION
**Pull request by adrien2p**

### CODE REVIEW

File: packages/core/framework/src/http/utils/validate-query.ts [hunk 1]
```diff
@@ -68,14 +68,19 @@ export function validateAndTransformQuery<TEntity extends BaseEntity>(
     next: NextFunction
   ) {
     try {
+      const restricted = req.restrictedFields.list()
```
Comment (olivermrbl) [inline packages/core/framework/src/http/utils/validate-query.ts]: thought: Not sure if we do this already, but we should probably add a strong recommendation in our docs that you should use `validateAndTransformQuery` for custom endpoints in the store domain

Comment (olivermrbl) [reply packages/core/framework/src/http/utils/validate-query.ts]: And to ensure best practices are followed, it should probably be the recommendation for any custom endpoint. Same with `validateAndTransformBody`

Comment (adrien2p) [reply packages/core/framework/src/http/utils/validate-query.ts]: yes in theory we are already recommending that especially if we talk about the query config

Comment (adrien2p) [reply packages/core/framework/src/http/utils/validate-query.ts]: and then if people want something very custom they can just reuse the tools that we are using under the hood in those middleware. But then it is up to the user to manage everything if they don't use our recommended way

Comment (adrien2p) [reply packages/core/framework/src/http/utils/validate-query.ts]: Apparently, only the *body is mentioned in the documentation, should we ask shahed to have a look?

Screenshot 2024-10-22 at 11 41 13

Comment (olivermrbl) [reply packages/core/framework/src/http/utils/validate-query.ts]: cc @shahednasser

Comment (adrien2p) [reply packages/core/framework/src/http/utils/validate-query.ts]: @olivermrbl do we also want to document ? or we do that later?
- `DEFAULT_STORE_RESTRICTED_FIELDS` from `framework/utils`
- `RestrictedFields` from `framework/http`
- medusa config `prjectConfig.http.restrictedFields`

File: packages/medusa/src/api/store/customers/query-config.ts [hunk 2]
```diff
@@ -38,6 +38,10 @@ export const defaultStoreCustomerAddressFields = [
 
 export const retrieveAddressTransformQueryConfig = {
   defaults: defaultStoreCustomerAddressFields,
+  allowed: [
```
Comment (olivermrbl) [inline packages/medusa/src/api/store/customers/query-config.ts]: todo: Can you add `validateAndTransformQuery` to the middlewares for the `GET /store/customers/me` endpoint? It is currently missing, so we can't query anything 😄

Comment (adrien2p) [reply packages/medusa/src/api/store/customers/query-config.ts]: it is present already

Screenshot 2024-10-22 at 14 44 14

Comment (olivermrbl) [reply packages/medusa/src/api/store/customers/query-config.ts]: Oh my bad, but looking at the wrong middelware config

Comment (olivermrbl) [inline packages/medusa/src/api/store/customers/query-config.ts]: todo: Can you add `allowed` to `retrieveTransformQueryConfig` in this file? 🙏

Comment (olivermrbl) [reply packages/medusa/src/api/store/customers/query-config.ts]: And remove `orders` here. This is for customer addresses

Comment (adrien2p) [reply packages/medusa/src/api/store/customers/query-config.ts]: done ✅

**Review by olivermrbl — APPROVED** [review]
LGTM!

**Pull request merged**